packages/connect/src/device/Device.ts

```diff
@@ -627,6 +647,23 @@ export class Device extends TypedEmitter<DeviceEvents> {
             }
         }
 
+        // TODO(karliatto): probably at this point we should send event to wait for it.
+        // const currentVersion = this.getVersion();
+        // TODO(karliatto): this is current version, we should get firmware for it before checking hash.
+        // TODO(karliatto): we should also get the latest version available in remote so we could
+        // request it also for later using it for offering update.
+        const releaseInfo = getReleaseInfo(this.features);
+        if (!releaseInfo) {
+            throw new Error('TODO');
+        }
+        this.firmwaresDfd = createDeferred();
+        this.emit(DEVICE.FIRMWARE_REQUEST, {
+            releases: [releaseInfo],
+            device: this.toMessageObject(),
+        });
+
+        await this.firmwaresDfd.promise;
```

Comment:
It looks like waiting here is not good, so I am searching for different approach to this.

Reply:
Right now I am making it here wait when device is connected but it creates some issues, so I am thinking that actually it could be done when Connect is initialized, it will request for all the firmwares already present and that information will be accessible from DataManager.

Since I realized there is not much value on waiting for device, since we can just ask for all the available firmwares in config directory. And then when they are needed we will just use them.

@mroz22 Where do you think would be best place to do that, so we can make connect initialization wait for that? Other option would be to get that information as a Connect initialization param.